openupgrade_scripts/scripts/delivery/17.0.1.0/pre-migration.py

```diff
@@ -47,6 +139,39 @@ def _delete_sql_constraints(env):
     )
 
 
+def _fill_sale_order_shipping_weight(env):
+    """Field `shipping_weight` on sale.order is a new stored field."""
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE sale_order
+        ADD COLUMN IF NOT EXISTS shipping_weight float8;
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        WITH weight_data AS (
+            SELECT
+                COALESCE(SUM(sol.product_uom_qty * p.weight), 0) AS shipping_weight,
+                sol.order_id
+            FROM sale_order_line sol
+            JOIN product_product p ON sol.product_id = p.id
+            JOIN product_template pt ON p.product_tmpl_id = pt.id
+            WHERE pt.type IN ('product', 'consu')
+                AND (sol.is_delivery IS NULL OR sol.is_delivery = FALSE)
```

Comment:
```suggestion
                AND NOT sol.is_delivery
```

Reply:
This is not working correctly.
With your proposal, the result is:
![image](https://github.com/user-attachments/assets/93e225cc-edf0-4a92-a75f-c5dfa9055510)

With the current code, the result is:
![image](https://github.com/user-attachments/assets/1721821e-6ea4-4800-9a7b-33375c91c152)

Because you have true, false, null:
![image](https://github.com/user-attachments/assets/2d2765b5-d2d0-40cb-a5b6-2fe59c84cbda)

Reply:
Oh sorry.

FYI Equivalent options:

```suggestion
                AND NOT COALESCE(sol.is_delivery, FALSE)
```
```suggestion
                AND sol.is_delivery IS DIFFERENT FROM TRUE
```
```suggestion
                AND (sol.is_delivery IS NULL OR sol.is_delivery = FALSE)
```